### ISSUE TITLE
mod_aggregate not working for list-form configuration

### ISSUE THREAD

**Issue by whiteinge** [saltstack/salt#18659]
The global setting for `state_aggregate` works when given `True` but not when given a list:

``` yaml
state_aggregate:
  - pkg
```

Tested on 2014.7.

**Comment by sebastian-cb** [comment]
This is still the case in _2015.5.5_:

>state_aggregate:
>  \- pkg

```
2015-12-14 14:32:35,010 [salt.loaded.int.module.cmdmod              ][INFO    ][23343] Executing command 'yum -y     install wget' in directory '/root'
2015-12-14 14:32:43,215 [salt.loaded.int.module.cmdmod              ][INFO    ][23343] Executing command 'yum -y     install innotop' in directory '/root'
```
>state_aggregate: True

```
2015-12-14 14:52:07,283 [salt.loaded.int.module.cmdmod              ][INFO    ][25434] Executing command 'yum -y     install wget innotop' in directory '/root'
```

**Comment by terminalmage** [comment]
Fixed by #41991.
